wire-ios-data-model/Source/MLS/MLSService.swift

```diff
@@ -828,32 +819,33 @@ public final class MLSService: MLSServiceInterface {
         guard let context = context else {
             return
         }
-        let pendingJoins = await context.perform { self.generatePendingJoins(in: context) }
-        for pendingJoin in pendingJoins {
-            try await joinByExternalCommit(groupID: pendingJoin.groupID)
-        }
 
-        groupsPendingJoin.removeAll()
-    }
-
-    private func generatePendingJoins(in context: NSManagedObjectContext) -> [PendingJoin] {
-        logger.info("generating list of groups pending join")
+        let pendingGroups = await context.perform { self.fetchPendingGroups(in: context) }
 
-        return groupsPendingJoin.compactMap { groupID in
+        logger.info("joining \(pendingGroups.count) group(s)")
 
-            guard let conversation = ZMConversation.fetch(with: groupID, in: context) else {
-                logger.warn("conversation not found for group (\(groupID.safeForLoggingDescription))")
-                return nil
+        await withTaskGroup(of: Void.self) { group in
+            for pendingGroup in pendingGroups {
+                group.addTask {
+                    do {
+                        try await self.joinByExternalCommit(groupID: pendingGroup)
+                    } catch {
+                        WireLogger.mls.error("Failed to join pending group (\(pendingGroup): \(error)")
+                    }
```

Comment:
**suggestion**: I guess the method does not throw then, we can remove the throws annotation.

What will happeb with the failed join, it will be tried forever at each start up?

Reply:
> What will happen with the failed join, it will be tried forever at each start up?

Yep, after each quick sync even.

I want to refactor joining MLS conversation some more and probably break it out of the `MLSService`. Currently there's nothing stopping it from starting multiple "joining pending conversations" operations if two quick sync happens in quick succession.

I'll create ticket for this

Reply:
> > suggestion: I guess the method does not throw then, we can remove the throws annotation.
> 
> 👍 I'll remove it

Had to keep because the fetch request can throw now.